NMS-11810: Database Reports documentation (#4663)
OpenNMS/opennms: docs/modules/operation/pages/database-reports/database.adoc

For more details on how JasperReports works, please refer to the Jaspersoft Studio link:http://community.jaspersoft.com/documentation/tibco-jaspersoft-studio-user-guide/v610/getting-started-jaspersoft-studio[official documentation].

[[ga-database-report-use-existing-report]]

== Using existing reports

OpenNMS provides the following sample report templates, located in `$\{OPENNMS_HOME}/etc/report-templates`, which you can customize to suit your needs:

[options="header, autowidth"]
|===
| Report Name            | Category     | Description

|  xref:operation:database-reports/templates/avail-summary.adoc[Availability Summary]
| Status
|Summary of statistics on outages and availability for user-specified surveillance category during a user-specified number of days.

| xref:operation:database-reports/templates/availability.adoc[Availability by Node]
| Status
| List of availability statistics for each node in user-specified surveillance category during a user-specified number of days.

> | xref:operation:database-reports/templates/availability.adoc[Default Calendar]

Reviewer: ```suggestion
| xref:operation:database-reports/templates/calendar.adoc[Default Calendar]
```

| Status
| Same availability information as Default Classic except that the two tables on daily availability are presented as a calendar.
This is not a JasperReport.

> | xref:operation:database-reports/templates/availability.adoc[Default Classic]

Reviewer: ```suggestion
| xref:operation:database-reports/templates/classic.adoc[Default Classic]
```

For more details on how JasperReports works, please refer to the Jaspersoft Studio link:http://community.jaspersoft.com/documentation/tibco-jaspersoft-studio-user-guide/v610/getting-started-jaspersoft-studio[official documentation].

[[ga-database-report-use-existing-report]]

> == Using existing reports

Reviewer 2: IMO, lines 208 and 209 are far enough out of date that they could safely be removed.

OpenNMS provides the following sample report templates, located in `$OPENNMS_HOME/etc/report-templates`, which you can customize to suit your needs:

[options="header, autowidth"]
|===
| Report Name            | Category     | Description

> |  xref:operation:database-reports/templates/avail-summary.adoc#avail-summary-report[Availability Summary]

Reviewer 2: Don't need to use the anchor in the link
```suggestion
|  xref:operation:database-reports/templates/avail-summary.adoc[Availability Summary]
```

| Status
| One page summary showing summary statistics on outages and availability for user-specified surveillance category during a user-specified number of days.

> | xref:operation:database-reports/templates/availability.adoc#availability-report[Availability by Node]

Reviewer 2: ```suggestion
| xref:operation:database-reports/templates/availability.adoc[Availability by Node]
```

| Status
| List of availability statistics for each node in user-specified category during a user-specified number of days.

| Default Calendar
| Unknown
| xxxx
// QUESTION: Need help getting info on this report.  It is not a JasperReport.

| Default Classic
| Unknown
| xxxx
// QUESTION: Need help getting info on this report.  It is not a JasperReport.

> | xref:operation:database-reports/templates/event.adoc#event-report[Event Analysis]

Reviewer 2: ```suggestion
| xref:operation:database-reports/templates/event.adoc[Event Analysis]
```

| Status
| Tables and charts showing different slices on the top 25 events over the past 30 days.

> | xref:operation:database-reports/templates/grafana.adoc#grafana-report[Grafana Dashboards]

Reviewer 2: ```suggestion
| xref:operation:database-reports/templates/grafana.adoc[Grafana Dashboards]
```

| Grafana
| Create Report from Grafana Dashboard, 3 options are provided.

> // ToDo: Get more detail on reports, find out why helper file with zoneid is not work; check to see if zoneid is depracated.

Reviewer 2: ```suggestion
// ToDo: Get more detail on reports, find out why helper file with zoneid is not work; check to see if zoneid is deprecated.
```

> | xref:operation:database-reports/templates/interface-avail.adoc#interface-avail-report[Interface Availability]

Reviewer 2: ```suggestion
| xref:operation:database-reports/templates/interface-avail.adoc[Interface Availability]
```

| Status
| Shows the availability of interfaces with outages within a time range.

> | xref:operation:database-reports/templates/maintenance-expired.adoc#maintenance-expired-report[Maintenance Contracts Expired]

Reviewer 2: ```suggestion
| xref:operation:database-reports/templates/maintenance-expired.adoc[Maintenance Contracts Expired]
```

| Asset Management
| Shows which assets with maintenance contracts are going to expire within a given time range.
// COMMENT: The name of this report is misleading to me.

> | xref:operation:database-reports/templates/maintenance-stragegy.adoc#maintenance-strategy-report[Maintenance Contracts Strategy]

Reviewer 2: ```suggestion
| xref:operation:database-reports/templates/maintenance-strategy.adoc#maintenance-strategy-report[Maintenance Contracts Strategy]
```

Reviewer 2: The file name needs to be renamed as well, to correct "strategy"

| Asset Management

> | Provides an overview of assets, showing how many within each age group have an active contract, no contract or an expired contract; and shows how many will be expiring in the next 12 months.

Author: That is the phrase used in the app.  Not my first choice.

> | xref:operation:database-reports/templates/morning.adoc#morning-report[Early Morning Report]

Reviewer 2: ```suggestion
| xref:operation:database-reports/templates/morning.adoc[Early Morning Report]
```

| Status
| Summary of status of network nodes over the past 24 hours.

> | xref:operation:database-reports/templates/response-summary.adoc#response-time-summary-report[Response Time Summary]

Reviewer 2: ```suggestion
| xref:operation:database-reports/templates/response-summary.adoc[Response Time Summary]
```

| Status
| Summary of response times across surveillance category.

> | xref:operation:database-reports/templates/response-time.adoc#response-time-report[Response Time by Node]

Reviewer 2: ```suggestion
| xref:operation:database-reports/templates/response-time.adoc[Response Time by Node]
```

| Status
| Response times by node.

| xref:operation:database-reports/templates/serial.adoc[Serial Interface Utilization Summary]
| Status

> | Summary of serial interface utilization.

Author: I think utilization is a term of art and it tends to refer to a measurement of capacity.  Even though they are clearly synonyms, it sounds weird to me to switch it.  It could be just me.  The example that comes to mind is if someone asks about your "storage utilization" that would prompt a percentage or a capacity description, but "storage use" is ambiguous as it could mean, I use the storage for storing Word files and spreadsheets but no videos or data files.

> | xref:operation:database-reports/templates/serial.adoc#serial-interface-report[Serial Interface Utilization Summary]

Reviewer 2: ```suggestion
| xref:operation:database-reports/templates/serial.adoc[Serial Interface Utilization Summary]
```

> | xref:operation:database-reports/templates/snmp.adoc#snmp-report[SNMP Interface Availability]

Reviewer 2: ```suggestion
| xref:operation:database-reports/templates/snmp.adoc[SNMP Interface Availability]
```

| Status
| Status of SNMP interfaces over time range.

> | xref:operation:database-reports/templates/total-bytes.adoc#total-bytes-report[Total Bytes Transferred by Interface]

Reviewer 2: ```suggestion
| xref:operation:database-reports/templates/total-bytes.adoc[Total Bytes Transferred by Interface]
```

| Usage
| Bytes transferred by interface.

> | xref:operation:database-reports/templates/traffic-rates.adoc#traffic-rate-report[Average and Peak Traffic Rates for Nodes by Interface]

Reviewer 2: ```suggestion
| xref:operation:database-reports/templates/traffic-rates.adoc[Average and Peak Traffic Rates for Nodes by Interface]
```

It also shows how many will expire in the next 12 months.

| xref:operation:database-reports/templates/morning.adoc[Early Morning Report]
| Status
| Summary of network node status over the past 24 hours.

| xref:operation:database-reports/templates/response-summary.adoc[Response Time Summary]
| Status
| Summary of response times across surveillance category.

| xref:operation:database-reports/templates/response-time.adoc[Response Time by Node]
| Status
| How long it takes a node to respond, listed by node.

| xref:operation:database-reports/templates/serial.adoc[Serial Interface Utilization Summary]
| Status
| Summary of serial interface utilization.

| xref:operation:database-reports/templates/snmp.adoc[SNMP Interface Availability]
| Status
| Status of SNMP interfaces over time range.

| xref:operation:database-reports/templates/total-bytes.adoc[Total Bytes Transferred by Interface]
| Usage
| Bytes transferred by interface.

| xref:operation:database-reports/templates/traffic-rates.adoc[Average and Peak Traffic Rates for Nodes by Interface]
| Usage
| Average and peak traffic rates for nodes by interface.

|===

> [ga-database-report-add-custom-report]]

Reviewer 2: ```suggestion
[[ga-database-report-add-custom-report]]
```

[[ga-database-reports-access-performance-data]]
== Accessing performance data

To access performance data within reports, we created a custom Measurement Datasource that lets you query the Measurements API and process the returned data in your reports.

> Please refer to the link:http://docs.opennms.org/opennms/branches/develop/guide-development/guide-development.html#_measurements_api[official Measurements API documentation] on how to use the Measurements API.

Reviewer 2: ```suggestion
Please refer to the https://docs.opennms.com/horizon/latest/development/rest/measurements.html[official Measurements API documentation] on how to use the Measurements API.
```
